File: Longest Palindromic Substring.java
/* Given a string S, find the longest palindromic substring in S. Substring of string S: S[ i . . . . j ] where 0 ≤ i ≤ j < len(S). 
Palindrome string: A string which reads the same backwards. 
More formally, S is palindrome if reverse(S) = S. Incase of conflict, return the substring which occurs first ( with the least starting index ).
*/



//User function Template for Java

class Solution{
    int max=0,start=0,end=0;
    String longestPalindrome(String S){
        // code here
        int n=S.length();
        for(int i=0;i<n;i++)
        {
            for(int j=i;j<n;j++)
            {
                if(isPalidrome(S,i,j))
                {
                    if((j-i)+1>max)
                    {
                        max=(j-i)+1;
                        start=i;
                        end=j;
                    }
                }
            }
        }
        return S.substring(start,end+1);
        
    }
    static boolean isPalidrome(String s,int left,int right)
    {
        while(left<right)
        {
            if(s.charAt(left)!=s.charAt(right))
            {
                return false;
            }
            else
            {
                left++;
                right--;
            }
        }
        return true;
    }
}

File: Longest Palindromic Substring.py
''' Given a string S, find the longest palindromic substring in S. Substring of string S: S[ i . . . . j ] where 0 ≤ i ≤ j < len(S). Palindrome string: A string which reads the same backwards. More formally, S is palindrome if reverse(S) = S. 
Incase of conflict, return the substring which occurs first ( with the least starting index ).'''

#User function Template for python3

class Solution:
    def longestPalindrome(self, a):
        # code here
        res=""
        resLen=0
        n=len(a)
        for i in range(n):
            
            #odd index
            l=i
            r=i
            
            while l>=0 and r<n and a[l]==a[r]:
                if r-l+1>resLen:
                    res=a[l:r+1]
                    resLen=r-l+1
                l-=1
                r+=1
            
            l=i
            r=i+1
            
            while l>=0 and r<n and a[l]==a[r]:
                if r-l+1>resLen:
                    res=a[l:r+1]
                    resLen=r-l+1
                l-=1
                r+=1
        return res
            
            
